python/foglamp/services/south/ingest.py

```diff
@@ -529,12 +538,10 @@ def is_available(cls) -> bool:
         # Comment out to test IntegrityError
         # key = '123e4567-e89b-12d3-a456-426655440000'
 
-        # Wait for an empty slot in the list
-        while not cls.is_available():
-            cls._readings_lists_not_full.clear()
-            await cls._readings_lists_not_full.wait()
-            if cls._stop:
-                raise RuntimeError('The South Service is stopping')
+        # If an empty slot is not available, discard the reading
+        if not cls.is_available():
+            cls.increment_discarded_readings()
```

Comment:
this `increment_discarded_readings` is okay, but why removed the clear and wait event. and  if stop why not the RuntimeError  `RuntimeWarning` actually

Reply:
clear and wait event handling has become now redundant  as we are not waiting for the empty slot now and simply discard the reading if queue is full. Secondly, RuntimeWarning, if raised, will require to handled somewhere. This method is fired by two callers - a) South server poll task, and b) async plugin. Both these callers do not have anything useful to do if this RuntimeWarning is raised. Besides, syslog error message is always generated when all ingest queues become full. Hence not provided.